test/intl402/DateTimeFormat/formatToParts/main.js

```diff
@@ -0,0 +1,46 @@
+// Copyright 2016 Mozilla Corporation. All rights reserved.
+// This code is governed by the license found in the LICENSE file.
+
+/*---
+description: Tests for existance and behavior of Intl.DateTimeFormat.prototype.formatToParts
+includes: [compareArray.js]
+---*/
+
+
+function assertArray(l, r) {
+  assert(compareArray(l, r), r);
+}
+
+function reduce(parts) {
+  return parts.map(part => part.value).join('');
+}
+
+function compareFTPtoFormat(locales, options, value) {
+  const dtf = new Intl.DateTimeFormat(locales, options);
+  assert.sameValue(
+    dtf.format(value),
+    reduce(dtf.formatToParts(value)),
+    `Expected the same value for value ${value},
+     locales: ${locales} and options: ${options}`
```

Comment:
can we avoid using DateTimeFormat#format to compare the values? In any other case, I believe it would be good to have a minimum check for type and value for each part.